specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/workspaceRP.json

```diff
@@ -3409,6 +3409,14 @@
             "read"
           ]
         },
+        "serverlessComputeCustomSubnet": {
+          "description": "The ARM resource ID for the subnet under which to create serverless compute clusters",
+          "type": "string"
+        },
+        "serverlessComputeNoPublicIP": {
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].
Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean.

Reply:
We want to align this property with an existing property [enableNodePublicIP](https://github.com/ac923/azure-rest-api-specs-arm/blob/mls-wsvnet-08-01-preview/specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json#L1971C15-L1971C15) which is a Boolean therefore we don't want to update the format to an Enum.